lib/mayaUsd/render/MaterialXGenOgsXml/libraries/1.38.7/mx_lighting_maya_v1.glsl

```diff
@@ -0,0 +1,27 @@
+// This is the same algorithm as found in libraries\pbrlib\genglsl\lib\mx_environment_prefilter.glsl
+// but adjusted for Maya. At this time we will compute a roughness based on the radiance and
+// irradiance samples, so materials with small amount of roughness will look wrong.
+//
+// A more precise roughness computation can be done using Maya samplers, but this requires
+// knowing that the Maya sampling functions are there, otherwise compilation will fail unless
+// there is an IBL active in the Maya lighting.
+
+#include "libraries/pbrlib/genglsl/lib/mx_microfacet_specular.glsl"
+
+vec3 mx_environment_irradiance(vec3 N)
+{
+    return g_diffuseI;
+}
+
+vec3 mx_environment_radiance(vec3 N, vec3 V, vec3 X, vec2 alpha, int distribution, FresnelData fd)
+{
+    N = mx_forward_facing_normal(N, V);
+    vec3 L = reflect(-V, N);
+    float NdotV = clamp(dot(N, V), M_FLOAT_EPS, 1.0);
+    float avgAlpha = mx_average_alpha(alpha);
+    vec3 F = mx_compute_fresnel(NdotV, fd);
+    float G = mx_ggx_smith_G2(NdotV, NdotV, avgAlpha);
+    vec3 comp = mx_ggx_energy_compensation(NdotV, avgAlpha, F);
+    vec3 Li = mix(g_specularI, g_diffuseI, avgAlpha);
+    return Li * F * G * comp;
+}
```

Comment:
This is a backup of the file one level above this and will work with MaterialX 1.38.4 to 1.38.7.